internal/aws/xray/telemetry/sender.go

```diff
@@ -161,34 +163,42 @@ func (p envMetadataProvider) get() string {
 }
 
 type ec2MetadataProvider struct {
-	client      *ec2metadata.EC2Metadata
-	metadataKey string
+	clientIMDSV2Only     *imds.Client
+	clientIMDSV1Fallback *imds.Client
+	metadataKey          string
 }
 
 func (p ec2MetadataProvider) get() string {
-	var metadata string
-	if result, err := p.client.GetMetadata(p.metadataKey); err == nil {
-		metadata = result
+	result, err := p.clientIMDSV2Only.GetMetadata(context.Background(), &imds.GetMetadataInput{Path: p.metadataKey})
+	if err != nil {
+		result, err = p.clientIMDSV1Fallback.GetMetadata(context.Background(), &imds.GetMetadataInput{Path: p.metadataKey})
+		if err != nil {
+			return ""
```

Comment:
All of these implementations swallow the error. Is there no way to bubble this up for debugging?

Reply:
I was trying to match functionality. I can add a warn log line.

Reply:
Added the warn line